CDCgov/prime-reportstream
# Experience/5441 valueset UI index page

This PR implements the first draft of the Valuesets Editor UI - Index page! Big props to @ahay-agile6 and @doug-s-nava who mobbed this with me.

Test Steps:
1. Login to RS
2. Navigate manually to /admin/value-sets in the URL (no navigation exists yet, since this is a work in progress)
3. Observe the list of valuesets

## Changes
- Changed the template/placeholder content of ValueSetsIndex.tsx and ValueSetsIndex.test.tsx to populate/test for actual content
- Added a hook (UseLookupTable.ts) and a network layer (LookupTableAPI.ts) with unit tests and a mock server (LookupTableMockServer.ts)
- Screenshot:
![image](https://user-images.githubusercontent.com/92405130/172933367-7f4f3bb3-1fb3-4fda-9096-168b3c81903a.png)
NOTE: Some artifacts such as the legend (above the header row) and the Add item button will _not_ ultimately be on this page, but will be moved to the Details page when it is developed in #5442 . Also, the Edit buttons will eventually navigate to the Details page- right now they do nothing, which is expected.

## Checklist

### Testing
- [ x ] Tested locally?
- [ x ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ x ] Added tests?

## Linked Issues
- Addressed #5441 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #5442 #5443
